rorapi/v2/serializers.py

```diff
@@ -87,3 +88,15 @@ class MatchedOrganizationSerializer(serializers.Serializer):
 class MatchingResultSerializer(serializers.Serializer):
     number_of_results = serializers.IntegerField()
     items = MatchedOrganizationSerializer(many=True)
+
+
+class ClientSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Client
+        fields = ['email', 'name', 'institution_name', 'institution_ror', 'country_code', 'ror_use']
+
+    def validate_email(self, value):
+        """Validate the email format and ensure it's unique."""
+        if Client.objects.filter(email=value).exists():
+            raise serializers.ValidationError("A client with this email already exists.")
+        return value
```

Comment:
Additional validation/sanitization of 'name', 'institution_name', 'institution_ror', 'country_code', and 'ror_use' fields is needed, per the spec https://docs.google.com/document/d/1ycUqXHqFZwejpkAsNLvJQCTKEsXN0-PZyfYL4QtYF-k/edit?tab=t.0#heading=h.lsgu7mypl9p2. 
- All fields should be sanitized to strip HTML/other potentially malicious chars and validated for length.  
- ROR ID formatting should be checked and converted to https://ror.org/XXXXX if needed
- Country code should be checked to ensure it's a valid ISO 3166-1 alpha-2 country code

This "pre-validation" could optionally be done at the view stage rather than the serializer, but I know it's a little more  Django-style to put it in the serializer.

Reply:
In general, `bleach` or `nh3` are common packages used in python for stripping out html.